pkg/client/issuecredential/doc.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package issuecredential provides support for the Issue Credential Protocol 2.0:
+// https://github.com/hyperledger/aries-rfcs/blob/master/features/0453-issue-credential-v2/README.md.
+//
+// Formalizes messages used to issue a credential. The protocol is responsible for orchestrating
+// the message flow according to the RFC.
+//
+// 1. Create your client:
+//
+// 	client, err := issuecredential.New(ctx)
+// 	if err != nil {
+// 	 panic(err)
+// 	}
+//
+// 2. Register an action event channel.
+//
+// 	actions := make(chan service.DIDCommAction)
+// 	client.RegisterActionEvent(actions)
+//
+// 3. Handle incoming actions.
+//
+//  for {
+//    select {
+//      case event := <-actions:
+//        piid := e.Properties.All()["piid"].(string)
+//
+//        if event.Message.Type() == presentproof.ProposeCredentialMsgType {
+//          // If Issuer is willing to accept the proposal.
+//          client.AcceptProposal(piid, &OfferCredential{})
+//          // If Issuer is not willing to accept the proposal.
+//          client.DeclineProposal(piid, reason)
+//        }
+//
+//        if event.Message.Type() == presentproof.OfferCredentialMsgType {
+//          // If Holder is willing to accept the offer.
+//          client.AcceptOffer(piid)
+//          // If Holder wants to counter an offer they received with a proposal.
+//          client.NegotiateProposal(piid, &ProposeCredential{})
+//          // If Holder is not willing to accept the offer.
+//          client.DeclineOffer(piid, reason)
+//        }
+//
+//        if event.Message.Type() == presentproof.RequestCredentialMsgType {
+//          // If Issuer is willing to accept the request.
+//          client.AcceptRequest(piid, &IssueCredential{})
+//          // If Issuer is not willing to accept the request.
+//          client.DeclineRequest(piid, reason)
+//        }
+//        if event.Message.Type() == presentproof.IssueCredentialMsgType {
+//          // If Holder is willing to accept the credentials.
+//          client.AcceptCredential(piid, names)
+//          // If Holder is not willing to accept the credentials.
+//          client.DeclineCredential(piid, reason)
+//        }
+//
+//        if event.Message.Type() == presentproof.ProblemReportMsgType {
+//          // Problem report message is triggered to notify client about the error.
+//          // In that case, there is only one option - accept it.
+//          client.AcceptProblemReport(piid)
```

Comment:
What happens when the problem report is ignored instead of accepted?

Reply:
@troyronda the protocol will not go to the final state. It is applicable to other actions as well.